noir/noir-repo/compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -41,16 +42,28 @@ pub fn collect_defs(
     module_id: LocalModuleId,
     crate_id: CrateId,
     context: &mut Context,
+    macro_processors: &[&dyn MacroProcessor],
```

Comment:
Why was this added?

Reply:
After discussing with @TomAFrench and @jfecher, it was agreed to keep this change in order to be able to process the whole AST in the first pass and not have to resort to the `def_collector` whenever possible. This is temporary anyways, until the new metaprograming feature is available.